### PR TITLE
Ajay tripathy add secrets

### PR DESCRIPTION
**Pull request by AjayTripathy**

### CODE REVIEW

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -419,10 +420,13 @@ serviceAccount:
 #  azureSubscriptionID: 0bd50fdf-c923-4e1e-850c-196dd3dcc5d3
 #  azureClientID: f2ef6f7d-71fb-47c8-b766-8d63a19db017
 #  azureTenantID: 72faf3ff-7a3f-4597-b0d9-7b0b201bb23a
+#  azureClientPassword: fake key # Only use if your values.yaml are stored encrypted. Otherwise provide an existing secret via serviceKeySecretName
 #  discount: "" # percentage discount applied to compute
 #  negotiatedDiscount: "" # custom negotiated cloud provider discount
 #  defaultIdle: false
-#  serviceKeySecretName: ""
+#  serviceKeySecretName: "" # Use an existing AWS or Azure secret with format as in aws-service-key-secret.yaml or azure-service-key-secret.yaml 
+#  createServiceKey: true # Creates a service key
```
Comment (dwbrown2) [inline cost-analyzer/values.yaml:428]: Is this field needed? Can we not just create the secret if a password is provided and no secret name is provided?

File: cost-analyzer/values.yaml [hunk 2]
```diff
@@ -419,10 +420,13 @@ serviceAccount:
 #  azureSubscriptionID: 0bd50fdf-c923-4e1e-850c-196dd3dcc5d3
 #  azureClientID: f2ef6f7d-71fb-47c8-b766-8d63a19db017
 #  azureTenantID: 72faf3ff-7a3f-4597-b0d9-7b0b201bb23a
+#  azureClientPassword: fake key # Only use if your values.yaml are stored encrypted. Otherwise provide an existing secret via serviceKeySecretName
 #  discount: "" # percentage discount applied to compute
 #  negotiatedDiscount: "" # custom negotiated cloud provider discount
 #  defaultIdle: false
-#  serviceKeySecretName: ""
+#  serviceKeySecretName: "" # Use an existing AWS or Azure secret with format as in aws-service-key-secret.yaml or azure-service-key-secret.yaml
```
Comment (dwbrown2) [inline cost-analyzer/values.yaml:427]: how do you feel about adding existing to this name? e.g. existingServiceKeySecretName or serviceKeyExistingSecret?

File: cost-analyzer/values.yaml [hunk 3]
```diff
@@ -419,10 +420,13 @@ serviceAccount:
...
 #  azureClientID: f2ef6f7d-71fb-47c8-b766-8d63a19db017
 #  azureTenantID: 72faf3ff-7a3f-4597-b0d9-7b0b201bb23a
+#  azureClientPassword: fake key # Only use if your values.yaml are stored encrypted. Otherwise provide an existing secret via serviceKeySecretName
 #  discount: "" # percentage discount applied to compute
 #  negotiatedDiscount: "" # custom negotiated cloud provider discount
 #  defaultIdle: false
-#  serviceKeySecretName: ""
+#  serviceKeySecretName: "" # Use an existing AWS or Azure secret with format as in aws-service-key-secret.yaml or azure-service-key-secret.yaml 
+#  createServiceKey: true # Creates a service key 
+#  createdServiceKeyType: AZURE # AWS or AZURE
```
Comment (dwbrown2) [inline cost-analyzer/values.yaml:429]: is this field necessary?

Comment (AjayTripathy) [reply cost-analyzer/values.yaml:429]: Yeah, unfortunately since the keys have different formats.

Comment (dwbrown2) [reply cost-analyzer/values.yaml:429]: If we are creating the secret, can we not detect if either an AWS or Azure has been supplied and then use the proper format?

Comment (AjayTripathy) [reply cost-analyzer/values.yaml:429]: Oh, yeah fair point. I'll fix

Comment (AjayTripathy) [reply cost-analyzer/values.yaml]: Ready for another look.

File: cost-analyzer/values.yaml [hunk 4]
```diff
@@ -419,10 +420,12 @@ serviceAccount:
 #  azureSubscriptionID: 0bd50fdf-c923-4e1e-850c-196dd3dcc5d3
 #  azureClientID: f2ef6f7d-71fb-47c8-b766-8d63a19db017
 #  azureTenantID: 72faf3ff-7a3f-4597-b0d9-7b0b201bb23a
+#  azureClientPassword: fake key # Only use if your values.yaml are stored encrypted. Otherwise provide an existing secret via serviceKeySecretName
 #  discount: "" # percentage discount applied to compute
 #  negotiatedDiscount: "" # custom negotiated cloud provider discount
 #  defaultIdle: false
-#  serviceKeySecretName: ""
+#  serviceKeySecretName: "" # Use an existing AWS or Azure secret with format as in aws-service-key-secret.yaml or azure-service-key-secret.yaml. Leave blank if using createServiceKeySecret 
+#  createServiceKeySecret: true # Creates a secret representing your cloud service key based on data in values.yaml. If you are storing unencrypted values, add a secret manually
```
Comment (dwbrown2) [inline cost-analyzer/values.yaml:428]: This flag feels like less of a concern, but it also doesn't feel totally necessary. Can't we just create if a key password is provided and no existing service key secret name is supplied? 

We should also clarify what take precedence if both are supplied.

**Review by dwbrown2 — APPROVED** [review]

**Pull request merged**